pkg/runtime/core/core.go

```diff
@@ -31,12 +31,25 @@ import (
 func New(ctx context.Context, client client.Client, indexer client.FieldIndexer) (map[string]runtime.Runtime, error) {
 	registry := NewRuntimeRegistry()
 	runtimes := make(map[string]runtime.Runtime, len(registry))
-	for name, factory := range registry {
-		r, err := factory(ctx, client, indexer)
-		if err != nil {
-			return nil, fmt.Errorf("initializing runtime %q: %w", name, err)
+	for name, registrar := range registry {
+		for _, dep := range registrar.dependencies {
+			depRegistrar, depExist := registry[dep]
```

Comment:
How we can add dependency to the runtime which doesn't exist in `runtimes` object ?

Reply:
There are not such situation since we need to initialize Registry, first like https://github.com/kubeflow/trainer/pull/2476/files#diff-0fc1a848bc1fddc453493f88c49c0b301d7b18d2fce44e27397383e82b29889cR32

Reply:
in that case, why do you need this check ? `if depExist`

Reply:
Because we can store arbitrary literal in `dependencies`. If they register incorrect dependency name, null pointer will happen.